### PR TITLE
Filtered_kernel: Add Power_side_of_oriented_power_circle_2.h

### LINKED ISSUE #7719: Missing static filter for Power_side_of_oriented_power_circle_2
## Issue Details

I got a report that CGAL was not competitive at all for 2D power diagrams. I did not run any benchmark, but I notice that there is no static filter for Power_side_of_oriented_power_circle_2, which is likely to be where most of the time is spent. This predicate is simple enough that writing a static filter shouldn't be too hard?

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Fix issue

### Todo

- [x] Add profiler statement for the different branches
- [ ] Does it improve performance (Marc mentioned a user)
- [x] Check for dead code

## Release Management

* Affected package(s): Filtered_kernel
* Issue(s) solved (if any): fix #7719

### CODE REVIEW

File: Filtered_kernel/include/CGAL/Filtered_kernel/internal/Static_filters/Power_side_of_oriented_power_circle_2.h [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+                {
+                  upper_bound_1 = max1;
+                }
+            }
+          if( (max2 < lower_bound_1) )
+            {
+              lower_bound_1 = max2;
+            }
+          if( ((lower_bound_1 < 2.99168207048872973507e-74) || (max4 < 8.95016161088373414772e-148)) )
+            {
```
Comment (albert-github) [inline Filtered_kernel/include/CGAL/Filtered_kernel/internal/Static_filters/Power_side_of_oriented_power_circle_2.h:137]: Wouldn't it be better to have a const or a define for this to tell what the values mean / is the idea behind it?

File: Filtered_kernel/include/CGAL/Filtered_kernel/internal/Static_filters/Power_side_of_oriented_power_circle_2.h [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+          fit_in_double(p.weight(), pwt) &&
+          fit_in_double(q.x(), qx) && fit_in_double(q.y(), qy)       &&
+          fit_in_double(q.weight(), qwt) &&
+          fit_in_double(t.x(), tx) && fit_in_double(t.y(), ty)       &&
+          fit_in_double(t.weight(), twt)
+          )
+        {
+          CGAL_BRANCH_PROFILER_BRANCH_1(tmp);
+          double dpx;
+          dpx = (px - tx);
```
Comment (albert-github) [inline Filtered_kernel/include/CGAL/Filtered_kernel/internal/Static_filters/Power_side_of_oriented_power_circle_2.h:191]: Why not:
```
double dpx = px - tx;
```
so on one line

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-116

**Comment by afabri** [comment]
> @afabri I see a TODO list in the top message, with unfinished tasks.
> 
> Do you want this PR merged as it is?

I think we can merge it as is. Thanks.

**Pull request merged**